go/vt/wrangler/materializer.go

```diff
@@ -65,6 +65,37 @@ const (
 	createDDLAsCopyDropConstraint = "copy:drop_constraint"
 )
 
+// addTablesToVSchema adds tables to an (unsharded) vschema. Depending on copyAttributes It will also add any sequence info
+// that is associated with a table by copying it from the vschema of the source keyspace.
+// For a migrate workflow we do not copy attributes since the source keyspace is just a proxy to import data into Vitess
+// Todo: For now we only copy sequence but later we may also want to copy other attributes like authoritative column flag and list of columns
+func (wr *Wrangler) addTablesToVSchema(ctx context.Context, sourceKeyspace string, targetVSchema *vschemapb.Keyspace, tables []string, copyAttributes bool) error {
+	if targetVSchema.Tables == nil {
+		targetVSchema.Tables = make(map[string]*vschemapb.Table)
+	}
+	for _, table := range tables {
+		targetVSchema.Tables[table] = &vschemapb.Table{}
+	}
+
+	if copyAttributes { // if source keyspace is provided, copy over the sequence info.
+		srcVSchema, err := wr.ts.GetVSchema(ctx, sourceKeyspace)
+		if err != nil {
+			return err
+		}
+		if srcVSchema == nil {
+			return fmt.Errorf("no vschema found for source keyspace %s", sourceKeyspace)
+		}
```

Comment:
I wonder if this has to be an error? If you're using shard targeting everywhere then you probably don't have a vschema and there would be no sequence info to copy either. 

I'm not sure if this method would even be called in that case though as this is only called if `if !vschema.Sharded`

Reply:
Removed the check.